### DISCUSSION TITLE
使用ispdomain模式，某些国内原本能用的服务会变得不好用

### DISCUSSION THREAD

**Discussion by chris6868** [joyanhui/ikuai-bypass#44] [General]
**简单描述**
在ispdomain模式下，苹果的appstore软件下载更新，不管国区还是外区，在梯子不好的情况下都会变慢，也就是都走了梯子，微软也是这样，哪怕手动添加了apple和微软的域名强制走wan1，但是其实还是会默认走wan2

**运行环境**
爱快版本：免费版3.7.16 x64 Build202411141700
ikuai-bypass版本：v2.1.2-alpha1
ikuai-bypass运行环境：debian12
ikuai-bypass运行环境：x86
**日志代码或截图支持md格式**

务必提供 完整命令 配置文件 以及运行日志，否则很有可能无法判断问题

爱快本身的使用问题，还有网络top结构问题，在移步 爱快官方论坛或恩山或者discussions  https://github.com/joyanhui/ikuai-bypass/discussions  

>

**Comment by chris6868** [comment]
修改成默认线路走wan1，然后gfw线路走wan2，就解决这个问题了，并且很好用

**Comment by joyanhui** [comment]
关于爱快分流的优先级顺序可以看下爱快官方文档。

**Comment by chris6868** [comment]
> 关于爱快分流的优先级顺序可以看下爱快官方文档。

这和分流优先级无关，主要是apple或微软不在国内的服务，但是国内直连是很好用的，但是使用分流后默认走wan2，如果梯子设置的非中国ip走梯子就会出现这个问题了

**Comment by joyanhui** [comment]
爱快默认优先级规则是  域名分流规则> ip分流规则  同时域名分流规则列表中id号较大的（后添加点）优先级高于id小的

默认线路个人建议是wan2 ，然后打开可用性检测。
国内ip走wan1，如果有个别的国外站点需要走wan1 建议参考  `https://github.com/joyanhui/ikuai-bypass/blob/main/config.yml`

里面 tag: private_direct  的用法 自己维护一个分流规则，并放在其他域名分流规则的末端。

clash部分 本身就有第三方转换规则可以适配微软和apple的域名，即便爱快这边没有分流好走到了clash也会DIRECT

**Reply by refined-fish** [reply]
想跟大佬探讨一下🫡
因为维护一份国外规则比较麻烦，所以似乎默认线路设置为wan2或更好，国内IP分流到wan1，再配合域名分流做补充
在这种方案下，国内IP分流到wan1有两种办法一种是负载均衡靠自定义运营商，一种是靠端口分流的外网线路模式，这两种实现有没有什么具体的区别呢，目前只知道在爱快的分流优先级中有区别

**Reply by refined-fish** [reply]
还有一个疑问，就是爱快官方也提到了使用域名分流必须配合多线路DNS，我的理解这是为了让分流过去的域名能够正确获得对应线路的DNS解析结果（最优DNS），但是我们这种双wan架构，本质上只有一条ISP，是不是不设置也罢
（PS：因为我开启了爱快DNS代理，爱快的DNS上游是openwrt中的mihomo，此时内网所有DNS都由爱快代理，就算开启多线路DNS也是不生效的，因为爱快只会走默认线路）

**Reply by joyanhui** [reply]
关于部分国外ip的域名直连的问题，我上面已经说的很清楚。补充：你自己在爱快内添加一条直接走wan1的域名分流规则也不是很复杂的事情。而且即便是走到了wan2，还有clash的规则，如果有强迫症就自己维护一个在线规则添加到config.yml。

关于 ip分组+端口分流 VS ISP分流+域名分流。个人建议有条件的选后者，没条件的选前者。条件指的：原来是单线ISP，并且在爱快内可以做3个网口（wan1 wan2  lan1 虚拟或者物理）。


最后的那个问题，你说的不设置也罢 是指的不设置什么 没搞明白。 至于和多线路dns的关系，不明白您基于什么方面的考量觉得有关系。我应该也没明白你的意思哈。
域名解析结果返回最优线路的问题 是域名的DNS服务器根据来源ip处理的。和运营商的dns还有共用dns无任何关系。

通常来说 dns 部分 只需要在 爱快的DNS代理配置1-2个国内的dot/doh就可以了，在clash内一般用一个好一点的配置模板就不用单独处理。如果有较高的要求，建议自建mosdns分流，然后配置为爱快DNS代理的上游和clash的DNS即可。

**Reply by joyanhui** [reply]
mosdns的配置和docker 可以参考我的另外一个仓库 https://github.com/joyanhui/mosdns

**Reply by refined-fish** [reply]
> 你说的不设置也罢 是指的不设置什么 没搞明白。

意思就是说，ISP分流下会有双wan，但实际上只有一个wan，所以多线DNS设置于不设置区别不大（爱快本意是多个ISP分流给对应的ISP优选DNS）。不过我看了你的答复我也理解了，DNS请求会根据源IP响应，所以你的意思是就算真的是多条ISP，其实不设置多线DNS也没啥大问题（那这么说感觉爱快的多线DNS没多大意义....）

>建议自建mosdns分流

这个对比mihomo的内建DNS有啥优势吗，mihomo内建DNS也可以分流，直接引用GEO数据库啥的，还可以设置DNS请求是否走代理来完全避免污染。
